choreolib/src/main/native/include/choreo/lib/Choreo.h

```diff
@@ -0,0 +1,106 @@
+   *
+   * @param trajName the path name in Choreo, which matches the file name in the
+   * deploy directory. Do not include ".traj" here.
+   * @return the loaded trajectory, will throw runtime error if the file doesn't
+   * exist
+   */
+  static ChoreoTrajectory GetTrajectory(std::string trajName);
+
+  /**
+   * Creates a CommandPtr that commands your drivebase to follow a Choreo
+   trajectory
+   *
+   * @param trajectory a ChoreoTrajectory to follow
+   * @param poseSupplier a function that returns a Pose2d of the robots current
+   * position
+   * @param xController a PIDController that controls the feedback on the global
+   * x position of the robot
+   * @param yController a PIDController that controls the feedback on the global
+   * y position of the robot
+   * @param rotationController a PIDController that controls the feedback on the
+   * global heading of the robot
+   * @param outputChassisSpeeds a function that consuming the calculated robot
+   * relative ChassisSpeeds
+   * @param useAllianceColor if true, mirror the trajectory along the midline of
+   * the field
+   * @param requirements the frc2::Requirements of the command
+   * @return an frc2::CommandPtr containing the command that will command your
+   * drivebase to follow a trajectory
+   */
+  static frc2::CommandPtr ChoreoSwerveCommandFactory(
+      ChoreoTrajectory trajectory, std::function<frc::Pose2d()> poseSupplier,
+      frc::PIDController xController, frc::PIDController yController,
+      frc::PIDController rotationController,
+      std::function<void(frc::ChassisSpeeds)> outputChassisSpeeds,
+      bool useAllianceColor, frc2::Requirements requirements = {});
+
+  /**
+   * Creates a CommandPtr that commands your drivebase to follow a Choreo
+   trajectory
+   *
+   * @param trajectory a ChoreoTrajectory to follow
+   * @param poseSupplier a function that returns a Pose2d of the robots current
+   * position
+   * @param controller a ChoreoControllerFunction that handles the feedback of
+   * the robots position
+   * @param outputChassisSpeeds a function that consuming the calculated robot
+   * relative ChassisSpeeds
+   * @param useAllianceColor if true, mirror the trajectory along the midline of
+   * the field
+   * @param requirements the frc2::Requirements of the command
+   * @return an frc2::CommandPtr containing the command that will command your
+   * drivebase to follow a trajectory
+   */
+  static frc2::CommandPtr ChoreoSwerveCommandFactory(
+      ChoreoTrajectory trajectory, std::function<frc::Pose2d()> poseSupplier,
+      ChoreoControllerFunction controller,
+      std::function<void(frc::ChassisSpeeds)> outputChassisSpeeds,
+      bool useAllianceColor, frc2::Requirements requirements = {});
+
+  /**
+   * Creates a ChoreoControllerFunction handles the feedback of the drivebase
+   * position
+   * @param xController a PIDController that controls the feedback on the global
+   * x position of the robot
```

Comment:
Indent Doxygen tag continuations two spaces. Spotless does this for Java, but we don't have a tool for it in C++.